lib/iris/cube.py

```diff
@@ -1694,7 +1695,8 @@ def data(self):
                 msg = msg.format(self.shape, data.dtype)
                 raise MemoryError(msg)
             # Unmask the array only if it is filled.
-            if isinstance(data, np.ndarray) and ma.count_masked(data) == 0:
+            if (isinstance(data, np.ma.masked_array) and
+                    ma.count_masked(data) == 0):
```

Comment:
@pp-mo It's not possible to put a masked array into a `dask.array.Array` (well, certainly one that **does** have its mask set) ... so is this really still valid in the new `dask` world?

Are you imagining that a user has a non-masked masked array wrapped up in a `dask.array.Array` ?